src/model.ts

```diff
@@ -0,0 +1,189 @@
+       *
+       * @method Model#get:
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {Model} callback.model The {@link Model}.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.get((err, model2, apiResponse) => {
+       *   // `model.metadata` has been populated.
+       * });
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [model2, apiResponse] = await model.get();
+       */
+      get: true,
+
+      /**
+       * Return the metadata associated with the Model.
+       *
+       * @see [Models: get API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/get}
+       *
+       * @method Model#getMetadata
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {object} callback.metadata The metadata of the Model.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * model.getMetadata((err, metadata, apiResponse) => {});
+       *
+       * @example <caption>If the callback is omitted we'll return a Promise
+       * </caption>
+       * const [metadata, apiResponse] = await model.getMetadata();
+       */
+      getMetadata: true,
+
+      /**
+       * @see [Models: patch API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/rest/v2/models/patch}
+       *
+       * @method Model#setMetadata
+       * @param {object} metadata The metadata key/value object to set.
+       * @param {function} [callback] The callback function.
+       * @param {?error} callback.err An error returned while making this
+       *     request.
+       * @param {object} callback.metadata The updated metadata of the Model.
+       * @param {object} callback.apiResponse The full API response.
+       * @returns {Promise}
+       *
+       * @example
+       * const {BigQuery} = require('@google-cloud/bigquery');
+       * const bigquery = new BigQuery();
+       * const dataset = bigquery.dataset('my-dataset');
+       * const model = dataset.model('my-model');
+       *
+       * const metadata = {
+       *   friendlyName: 'thebestmodelever'
```

Comment:
Can it be "theBestModelEver", or is all lowercase really the deal?

Reply:
We can do that!